Issue #1991  import of spambayes classifier

vezult opened the pull request:
Added spambayes classifier to the mailpile tree.

  * Imported only the classifier and its dependancies
  * Removed external URL retrieval code
  * Modified code to be python2.7/3.x compatible
  * Generally tried to modify everything as little as possible

vezult commented:
Testing:
    * All spambayes tests pass
    * Autotagging remains enabled even in a virtualenv without an external spambayes module
    * No debug/error messages from the autotag plugin
    * New messages show up in the spam folder

BjarniRunar commented:
Thank you for working on this! I'm going to finish my current round of Autocrypt and PGP-key related work before I dive into this, so there will be a bit of a delay before this gets merged.

jonathan-s commented on mailpile/spambayes/OptionsClass.py:
```diff
@@ -89,6 +89,8 @@
     except ImportError:
         import io as StringIO
 
+    xrange = range
```
Not sure how many similar cases like this there are. But an idea is to have a `py3.py` (or other naming) file and from that file you import all the the methods that have changed. 

That way it'll all be collected in one single file rather than a bit spread out.

vezult replied:
The proper thing to do would probably be to use [six](https://pythonhosted.org/six/) which is listed in requirements-with-deps.txt. Seems like there is a desire to avoid external dependencies as much as possible, so I don't know if it would be acceptable to turn that into a direct dependency or not. I'd be for it.

It appears it's only use in this case would produce a 128 integer array, and it would only happen on import. Using a generator there doesn't seem like a huge win. I could just convert it to use `range()`. My goal in all cases was to try to stay as close to the original as possible. I don't know how much that matters.

BjarniRunar replied:
There is a desire to avoid external dependencies that complicate packaging work. I don't think six is a problem, it should be everywhere by now.

jonathan-s commented:
Personally I would also squash all the python compability changes into one. Ie having "remove url retrieval" and "import spambayes classifier" as separate commits is fine, but the granular commits for each of the compability changes I would squash.

vezult commented:
OK, I've:
  * squashed python3 compatibility commits
  * switched to use of [six](https://pythonhosted.org/six/) for handing between python2/3 (@BjarniRunar opinions?)
  * removed the spambayes dependency on [python-lockfile](https://pypi.org/project/lockfile/) in favor of [fasteners](https://pypi.org/project/fasteners/), which appears to already be a direct dependency

BjarniRunar commented:
I have merged this in my local tree, once I've verified it seems to behave, I'll push to master. Thanks again for doing this!

BjarniRunar commented:
This is merged and public. I guess my rebasing confused GitHub, so it didn't recognize the merge. But it's done. Thank you!
